### DISCUSSION TITLE
ContentItems not persisted

### DISCUSSION THREAD

**Discussion by ricocsharp** [OrchardCMS/OrchardCore#12736] [Q&A]
Hello, I'm creating ContentItems via code, there are no errors and debugging the creation Method  I can see all the info about the new contentItem, but  at the end this new ContentItem is not presisted in DB here is the fragment of code:

var contentItem = await _contentManager.NewAsync(productoContentType);
                                contentItem.Owner = User.Identity.Name;
                                contentItem.Alter<ProductoPart>(part =>
                                {
                                    part.Categoria = details.Categoria;
                                    part.Cup = details.Cup;
                                    part.Hsa = details.Hsa;
                                    part.Envase = details.Envase;
                                    part.Marca = details.Marca;
                                    part.Tipo = details.Tipo;
                                    part.NormaCajon = details.NormaCajon;
                                    part.NormaMC = details.NormaMC;
                                    part.PesoNeto = details.PesoNeto;
                                    part.VitolaGalera = details.VitolaGalera;
                                    part.VitolaSalida = details.VitolaSalida;
                                });
                                try
                                {
                                    await _contentManager.UpdateValidateAndCreateAsync(contentItem, VersionOptions.Draft);
                                    await _contentManager.PublishAsync(contentItem);
                                }
                                catch (Exception ex)
                                {
                                    
                                }

**Comment by jtkech** [comment]
Normally you should provide the type of your part `contentItem.Alter<TheTypeOfYourPart>(part =>`

**Reply by ricocsharp** [reply]
@jtkech thanks for your answer,  now i have made some changes and the code is this:

                             foreach (var details in productos)
                            {
                                    string DisplayText = details.Hsa;
                                    var contentItem = await _contentManager.NewAsync(productoContentType);
                                    contentItem.Owner = User.Identity.Name;
                                    contentItem.Alter<ProductoPart>(x =>
                                     {
                                         x.Alter<TextField>("Categoria", d => d.Text = details.Categoria);
                                         x.Alter<TextField>("Cup", d => d.Text = details.Cup);
                                         x.Alter<TextField>("Hsa", d => d.Text = details.Hsa);
                                         x.Alter<TextField>("Envase", d => d.Text = details.Envase);
                                         x.Alter<TextField>("Marca", d => d.Text = details.Marca);
                                         x.Alter<TextField>("Tipo", d => d.Text = details.Tipo);
                                         x.Alter<TextField>("NormaCajon", d => d.Text = details.NormaCajon);
                                         x.Alter<TextField>("NormaMC", d => d.Text = details.NormaMC);
                                         x.Alter<TextField>("PesoNeto", d => d.Text = details.PesoNeto);
                                         x.Alter<TextField>("VitolaGalera", d => d.Text = details.VitolaGalera);
                                         x.Alter<TextField>("VitolaSalida", d => d.Text = details.VitolaSalida);
                                     });                                
                                    contentItem.DisplayText = DisplayText;
                                    try
                                    {
                                        await _contentManager.UpdateValidateAndCreateAsync(contentItem, VersionOptions.Draft);
                                        await _contentManager.PublishAsync(contentItem);
                                    }
                             }
 now the ProductPart is persisted but with an Empty  ProductPart here is the value of the Content Field of the Document table in DB, note that ProductPart have null values in every property,

{"ContentItemId":"48tz5j3bwz36gt2dk20qmdsqnr","ContentItemVersionId":"4f4rmp2xqmycaygvrkq1q35cwd","ContentType":"Producto","DisplayText":"00.8016.22.1280","Latest":true,"Published":true,"ModifiedUtc":"2022-10-31T15:36:20.2555869Z","PublishedUtc":"2022-10-31T15:36:20.2585548Z","CreatedUtc":"2022-10-31T15:36:20.2555869Z","Owner":"javier.rico","Author":"javier.rico",

"ProductoPart":{"Cup":null,"Hsa":null,"Marca":null,"VitolaSalida":null,"VitolaGalera":null,"Categoria":null,"NormaMC":null,"NormaCajon":null,"PesoNeto":null,"Envase":null,"Tipo":null}}

as U can see in this snapshot of my debugging 
![image](https://user-images.githubusercontent.com/4331822/199051771-5c93887b-ba17-4fce-91c3-6acd44ba77f7.png)
 the details object is pupulated but the  asigment of it's value to the Part is not working Do u see Why is that?

thanks in advance for your response

Bests

**Reply by jtkech** [reply]
The code looks good, can you show us how your content type and part are defined.
